worlds/banjo_tooie/__init__.py

```diff
@@ -0,0 +1,1094 @@
+        elif itemname == itemName.NOTE_AS_USEFUL:
+            itemname = itemName.NOTE
+            if not hasattr(self.multiworld, "generation_is_fake"):
+                item_classification = ItemClassification.useful
+        elif itemname == itemName.DOUBLOON_AS_FILLER:
+            itemname = itemName.DOUBLOON
+            if not hasattr(self.multiworld, "generation_is_fake"):
+                item_classification = ItemClassification.filler
+        elif itemname == itemName.HEALTHUP and (
+            self.options.logic_type
+                == LogicType.option_easy_tricks or self.options.logic_type == LogicType.option_intended
+        ):
+            item_classification = ItemClassification.useful
+
+        banjoItem = all_item_table.get(itemname)
+        if not banjoItem:
+            raise Exception(f"{itemname} is not a valid item name for Banjo-Tooie")
+
+        if item_classification is None:
+            item_classification = self.get_classification(banjoItem)
+
+        if item_classification == ItemClassification.trap:
+            self.traps_in_pool += 1
+
+        if itemname == itemName.JIGGY:
+            self.jiggies_in_pool += 1
+        if itemname == itemName.NOTE:
+            self.notes_in_pool += 1
+        if itemname == itemName.DOUBLOON:
+            self.doubloons_in_pool += 1
+
+        created_item = BanjoTooieItem(itemname, item_classification, banjoItem.btid, self.player)
+        return created_item
+
+    def get_classification(self, banjoItem: ItemData) -> ItemClassification:
+        if banjoItem.btid is not None:
+            itemname = self.item_id_to_name[banjoItem.btid]
+
+            if itemname == itemName.PAGES:
+                if self.options.cheato_rewards:
+                    return ItemClassification.progression_deprioritized_skip_balancing
+                else:
+                    return ItemClassification.filler
+
+            if itemname == itemName.HONEY:
+                if self.options.honeyb_rewards:
+                    return ItemClassification.progression_deprioritized_skip_balancing
+                else:
+                    return ItemClassification.useful
+
+        if banjoItem.type not in (
+            ItemClassification.progression,
+            ItemClassification.progression_deprioritized_skip_balancing,
+            ItemClassification.useful,
+            ItemClassification.filler,
+            ItemClassification.trap
+        ):
+            raise Exception(f"{banjoItem.type} does not correspond to a valid item classification.")
+
+        return banjoItem.type
+
+    def create_event_item(self, name: str) -> Item:
+        item_classification = ItemClassification.progression
+        created_item = BanjoTooieItem(name, item_classification, None, self.player)
+        return created_item
+
+    def calculate_useful_filler(self, total, progression) -> tuple[int, int]:
+        # We want to split non-progressive items into two halfs
```

Comment:
I assume this comment means "non-progression" items, although I'm somewhat confused about the purpose of doing this. Is there a reason why some but not all of these items should be excludable?